src/command.ts

```diff
@@ -29,104 +36,116 @@ export type CypressConfigEnv = {
   visualRegressionFailSilently?: boolean
 }
 
-export type TakeScreenshotProps = {
-  path: string
-  size: number
-  dimensions: {
-    width: number
-    height: number
-  }
-  multipart: boolean
-  pixelRatio: number
-  takenAt: string
-  name: string
-  blackout: string[]
-  duration: number
-  testAttemptIndex: number
-}
+type TakeScreenshotProps = Partial<Cypress.ScreenshotOptions> & { path: string }
 
 /** Add custom cypress command to compare image snapshots of an element or the window. */
-function addCompareSnapshotCommand(screenshotOptions?: ScreenshotOptions): void {
+function addCompareSnapshotCommand(screenshotOptions?: Partial<Cypress.ScreenshotOptions & PluginOptions>): void {
   Cypress.Commands.add(
     'compareSnapshot',
     { prevSubject: ['optional', 'element'] },
     function (
-      subject: JQueryWithSelector<HTMLElement> | void,
+      subject: Cypress.JQueryWithSelector | void,
       name: string,
       commandOptions?: PluginCommandOptions
     ): Cypress.Chainable {
       if (name === undefined || name === '') {
         throw new Error('Snapshot name must be specified')
       }
-      let errorThreshold = 0
-      // prepare screenshot options
-      if (screenshotOptions?.errorThreshold !== undefined) {
-        errorThreshold = screenshotOptions?.errorThreshold
-      }
-      let regressionOptions: ScreenshotOptions
-      if (typeof commandOptions === 'object') {
-        regressionOptions = { ...screenshotOptions, ...commandOptions }
-        if (commandOptions.errorThreshold !== undefined) {
-          errorThreshold = commandOptions.errorThreshold
-        }
+
+      const realScreenshotOptions = { ...screenshotOptions }
```

Comment:
I can't figure out why you need to declare realScreenshotOptions, I don't see that you are mutating screenshotOptions

Reply:
`screenshotOptions` can be `undefined`, and if I do this, it assures at least `{}`. I'm not a fan of it also, open for suggestions

Reply:
I renamed it to `screenshotOptionsObject`, and left a comment to explain the purpose. Should be more friendly now